src/parser/jobs/ast/modules/ArcanaTracking/CritAndDHRates.tsx

```diff
@@ -0,0 +1,422 @@
+	private critPartyModifierCurrent: {[actor: Actor['id']]: {currentModifier: number}} = {}
+	private critEnemyModifierCurrent: {[actor: Actor['id']]: {currentModifier: number}} = {}
+	private DHPartyModifierCurrent: {[actor: Actor['id']]: {currentModifier: number}} = {}
+	private appliedStatuses: Array<{actor: Actor['id'], status: Status['id']}> = [] //reasoning is that bard's statuses are reapplied without removal. this is to catch those
+
+	//crit and DH Modifier variables
+	private critPartyModifiers: {[id: Status['id']]: {strength: number}} = {
+		[this.data.statuses.BATTLE_LITANY.id]: {strength: 0.1},
+		[this.data.statuses.DEVILMENT.id]: {strength: 0.2},
+		[this.data.statuses.THE_WANDERERS_MINUET.id]: {strength: 0.02},
+		[this.data.statuses.INNER_RELEASE.id]: {strength: 1},
+		//TO DO: these below moves only increase crit for the next weaponskill, not the oGCDs between
+		[this.data.statuses.REASSEMBLED.id]: {strength: 1},
+		[this.data.statuses.OPO_OPO_FORM.id]: {strength: 1},
+		[this.data.statuses.LIFE_SURGE.id]: {strength: 1},
+	}
+
+	private critEnemyModifiers: {[id: Status['id']]: {strength: number}} = {
+		[this.data.statuses.CHAIN_STRATAGEM.id]: {strength: 0.1},
+	}
+
+	private DHPartyModifiers: {[id: Status['id']]: {strength: number}} = {
+		[this.data.statuses.ARMYS_PAEON.id]: {strength: 0.03},
+		[this.data.statuses.BATTLE_VOICE.id]: {strength: 0.2},
+		[this.data.statuses.DEVILMENT.id]: {strength: 0.2},
+		[this.data.statuses.REASSEMBLED.id]: {strength: 1},
+		[this.data.statuses.INNER_RELEASE.id]: {strength: 1},
+	}
+
+	//to store the above crits, dh in an easy to call variable
+	private critPartyStatuses: Array<Status['id']> = [
+		this.data.statuses.BATTLE_LITANY.id,
+		this.data.statuses.DEVILMENT.id,
+		this.data.statuses.THE_WANDERERS_MINUET.id,
+		this.data.statuses.INNER_RELEASE.id,
+		this.data.statuses.REASSEMBLED.id,
+		this.data.statuses.OPO_OPO_FORM.id,
+		this.data.statuses.LIFE_SURGE.id,
+	]
+	private critEnemyStatuses: Array<Status['id']> = [
+		this.data.statuses.CHAIN_STRATAGEM.id,
+	]
+	private DHPartyStatuses: Array<Status['id']> = [
+		this.data.statuses.DEVILMENT.id,
+		this.data.statuses.REASSEMBLED.id,
+		this.data.statuses.INNER_RELEASE.id,
+		this.data.statuses.BATTLE_VOICE.id,
+		this.data.statuses.ARMYS_PAEON.id,
+	]
+
+	//these are abilities that will automatically crit or direct hit which will skew results and should automatically be removed from the respective DH or Crit calculations
+	private autoCritAbilities: Array<Action['id']> = [
+		this.data.actions.STARFALL_DANCE.id,
+		this.data.actions.CHAOTIC_CYCLONE.id,
+		this.data.actions.INNER_CHAOS.id,
+		this.data.actions.PRIMAL_REND.id,
+	]
+	private autoDHAbilities: Array<Action['id']> = [
+		this.data.actions.STARFALL_DANCE.id,
+		this.data.actions.CHAOTIC_CYCLONE.id,
+		this.data.actions.INNER_CHAOS.id,
+		this.data.actions.PRIMAL_REND.id,
+	]
```

Comment:
Wildfire needs some handling too. It can't crit or DH so the base damage is the only part that counts.

Reply:
Do you happen to know if wildfire is the only ability like this?

Reply:
I believe so yes

Reply:
Thanks! I added a section to handle this a bit generally in case this comes up again in the future
See below commit